fullstack-cert/python-projects/email-simulator/main.py

```diff
@@ -0,0 +1,113 @@
+import datetime
+
+class Email:
+    def __init__(self, sender, receiver, subject, body):
+        self.sender = sender
+        self.receiver = receiver
+        self.subject = subject
+        self.body = body
+        self.timestamp = datetime.datetime.now()
+        self.read = False
+
+    def mark_as_read(self):
+        self.read = True
+
+    def display_summary(self):
+        print(str(self))
+
+    def display_full_email(self):
+        self.mark_as_read()
+        print("\n--- Email ---")
+        print(f"From: {self.sender.name}")
+        print(f"To: {self.receiver.name}")
+        print(f"Subject: {self.subject}")
+        print(f"Received: {self.timestamp.strftime('%Y-%m-%d %H:%M')}")
+        print("Body:")
+        print(self.body)
+        print("------------\n")
+
+    def __str__(self):
+        status = "Read" if self.read else "Unread"
+        return f"[{status}] From: {self.sender.name} | Subject: {self.subject} | Time: {self.timestamp.strftime('%Y-%m-%d %H:%M')}"
+
+
+class Inbox:
+    def __init__(self):
+        self.emails = []
+
+    def receive_email(self, email):
+        self.emails.append(email)
+
+    def list_emails(self):
+        if not self.emails:
+            print("Your inbox is empty.\n")
+            return
+        print("\nYour Emails:")
+        for i, email in enumerate(self.emails, start=1):
+            print(f"{i}. {str(email)}")
+        print()
+
+    def read_email(self, index):
+        if not self.emails:
+            print("Inbox is empty.\n")
+        elif 0 <= index < len(self.emails):
+            self.emails[index].display_full_email()
+        else:
+            print("Invalid email number.\n")
```

Comment:
```suggestion
        try:
            self.emails[index].display_full_email()
        except IndexError:
            print("Invalid email number.\n")
```
In this way you are allowed to use negative indexing.